core/src/main/kotlin/org/evomaster/core/search/service/ExtraHeuristicsLogger.kt

```diff
@@ -21,45 +28,61 @@ class ExtraHeuristicsLogger {
     @PostConstruct
     private fun postConstruct() {
         if (config.writeExtraHeuristicsFile) {
-
-            val path = getFilePath()
-
-            Files.createDirectories(path.parent)
-
-            Files.deleteIfExists(path)
-            Files.createFile(path)
-
-            path.toFile().appendText("testCounter,actionId,value,id,type,objective,group\n")
+            val extraHeuristicsFilePath = getExtraHeuristicsFilePath()
+            setUpFilePath(extraHeuristicsFilePath)
+            extraHeuristicsFilePath.toFile().appendText("testCounter,actionId,value,id,type,objective,group\n")
         }
     }
 
-    private fun getFilePath() = Paths.get(config.extraHeuristicsFile).toAbsolutePath()
+    /**
+     * Creates parent path if it is not exists.
+     * And deletes the file if the file already exists.
+     */
+    private fun setUpFilePath(filePath: Path) {
+        Files.createDirectories(filePath.parent)
+        Files.deleteIfExists(filePath)
+        Files.createFile(filePath)
+    }
 
+    private fun getExtraHeuristicsFilePath() = Paths.get(config.extraHeuristicsFile).toAbsolutePath()
 
     fun writeHeuristics(list: List<HeuristicEntryDto>, actionId: Int) {
-        if (!config.writeExtraHeuristicsFile) {
-            return
-        }
+        if (config.writeExtraHeuristicsFile) {
```

Comment:
revert. usually it is better to return immediately on a negated condition, as it avoid a long nested block

Reply:
ok.